js/src/ui/Form/TypedInput/typedInput.js

```diff
@@ -26,30 +26,33 @@ import { fromWei, toWei } from '~/api/util/wei';
 import Input from '~/ui/Form/Input';
 import InputAddressSelect from '~/ui/Form/InputAddressSelect';
 import Select from '~/ui/Form/Select';
-import { ABI_TYPES } from '~/util/abi';
+import { ABI_TYPES, parseAbiType } from '~/util/abi';
 
 import styles from './typedInput.css';
 
 export default class TypedInput extends Component {
 
   static propTypes = {
     onChange: PropTypes.func.isRequired,
-    param: PropTypes.object.isRequired,
 
     accounts: PropTypes.object,
     error: PropTypes.any,
-    value: PropTypes.any,
-    label: PropTypes.string,
     hint: PropTypes.string,
-    min: PropTypes.number,
+    isEth: PropTypes.bool,
+    label: PropTypes.string,
     max: PropTypes.number,
-    isEth: PropTypes.bool
+    min: PropTypes.number,
+    param: PropTypes.oneOfType([
+      PropTypes.object,
+      PropTypes.string
+    ]).isRequired,
+    value: PropTypes.any
   };
 
   static defaultProps = {
     min: null,
     max: null,
-    isEth: false
+    isEth: null
```

Comment:
Ah yes missed that one. The idea is just that by default it's using the given param, but if `isEth` is passed, then it will show the `ETH` toggle. If `isEth` is true => default to `ETH` value, if not, default to integer value. Will change the PropTypes though